prime-router/src/main/kotlin/Element.kt

```diff
@@ -1250,6 +1280,47 @@ data class Element(
                 else -> value
             }
         }
+
+        /**
+         * Given [cleanedValue] this method checks to see if it is possibly a phone number in a safe
+         * way without blowing up all the processing of rows in a report. If the phone number is probably
+         * valid it returns null, as the `checkForErrors` method would expect, otherwise it returns an
+         * InvalidPhoneMessage
+         */
+        fun checkPhoneNumber(cleanedValue: String, fieldMapping: String): InvalidPhoneMessage? {
+            // use a quick regex to see if the normalized value contains something other than our
+            // expected values, and also use the `isPossibleNumber` method our phone number util gives us
+            if (
+                maybeAPhoneNumber.matchEntire(cleanedValue) != null &&
+                phoneNumberUtil.isPossibleNumber(cleanedValue, "US")
+            ) {
+                // attempt to parse the number. if it is parseable then we can return null and move
+                // on with our work, otherwise, return an InvalidPhoneMessage
+                val phoneNumber = tryParsePhoneNumber(cleanedValue)
+                if (phoneNumber != null)
+                    return null
+            }
+            // all attempts have failed. bad number
+            return InvalidPhoneMessage(cleanedValue, fieldMapping)
+        }
+
+        /**
+         * Given a nullable [cleanedValue] this method tries to parse a phone number in
+         * a safe way according to our four most common phone regions: US, Mexico, Canada, and Australia.
+         * If the number really can't be parsed at all (and the phoneNumberUtil is very lenient) then
+         * it is almost certainly not a phone number, so return null
+         */
+        fun tryParsePhoneNumber(cleanedValue: String?): Phonenumber.PhoneNumber? {
+            for (region in phoneRegions) {
```

Comment:
The comment references the "four most common phone regions" but the `phoneRegions` has about 50 entries, not four, so something isn't quite right here.